homeassistant/components/iqvia/sensor.py

```diff
@@ -94,14 +94,19 @@ def calculate_trend(indices):
 
 
 class ForecastSensor(IQVIAEntity):
-    """Define sensor related to forecast data."""
+    """Define sensor related to IQVIA forecast data."""
 
     async def async_update(self):
         """Update the sensor."""
         if not self._iqvia.data:
             return
 
-        data = self._iqvia.data[self._type].get('Location')
+        try:
+            data = self._iqvia.data[self._type].get('Location')
+        except AttributeError:
```

Comment:
I agree we don't want the traceback, but this error string isn't particularly useful.. how about:

```suggestion
        except AttributeError as err:
            _LOGGER.error("Unable to update data: %s", err)
```

Reply:
I think this problem should be handled closer to the source. It's not clear to me why we should accept an attribute error. It should probably be fixed instead to not generate that error.

Reply:
I agree with @MartinHjelmare: I'm concerned that this is merely masking an operational error. I'll dig in and see if I can find the root cause.

Reply:
Following up – the only time I can reproduce this error is when I'm completely disconnected from the internet. In that case, I think this fix might be appropriate (unless there's a way to halt a sensor's `async_update` upstream)?

Reply:
We should still avoid this as intended behavior... based on the traceback that @fabaff posted, it looks like `self._iqvia.data[TYPE_ALLERGY_INDEX]` is being set to a `ClientConnectorError`.  I assume the library is setting the error instance in the data when it fails -- it would be better to let it raise an error instead to make the interface less confusing.

Alternatively, you could add a gate to check that `self._iqvia.data[TYPE_ALLERGY_INDEX]` is a `dict` explicitly (I assume it's a dict):

```python 
disease_index = self._iqvia.data[TYPE_DISEASE_INDEX]
if not isinstance(disease_index, dict):
    _LOGGER.error("Unable to get data: %s", disease_index)
data = disease_index.get('Location')
```

Reply:
`pyiqvia` isn't doing this: it looks like it goes deeper to `aiohttp`. At any rate, I like that second suggestions.

CC: @fabaff

Reply:
I suggest we check if the result is an instance of an `Exception` as the last case here:
https://github.com/home-assistant/home-assistant/blob/2c0cf1abd9d93d79f706e30e9551ee6094d5c2af/homeassistant/components/iqvia/__init__.py#L138-L148